synapse/config/experimental.py

```diff
@@ -71,6 +71,9 @@ def read_config(self, config: JsonDict, **kwargs: Any) -> None:
         self.msc3720_enabled: bool = experimental.get("msc3720_enabled", False)
 
         # MSC2654: Unread counts
+        #
+        # Note that when this is enabled previously calculated push rules might
+        # be incorrect.
```

Comment:
Previously if you enabled this flag the counts would be 100% correct for all previous message. This is no longer true and it will only be correct from the time the flag is enabled.

I think this is OK as the feature is still experimental.